src/lib/y2users/linux/users_writer.rb

```diff
@@ -0,0 +1,366 @@
+      end
+
+      # Generates and returns the options expected by `useradd` for given user
+      #
+      # @param user [User]
+      # @return [Array<String>]
+      def useradd_options(user)
+        opts = {
+          "--uid"      => user.uid,
+          "--gid"      => user.gid,
+          "--shell"    => user.shell,
+          "--home-dir" => user.home,
+          "--comment"  => user.gecos.join(","),
+          "--groups"   => useradd_groups(user).join(",")
+        }
+        opts = opts.reject { |_, v| v.to_s.empty? }.flatten
+
+        if user.system?
+          opts << "--system"
+        else
+          opts.concat(create_home_options(user))
+        end
+
+        opts << user.name
+        opts
+      end
+
+      # Name of the secondary groups for useradd
+      #
+      # @return [Array<String>]
+      def useradd_groups(user)
+        names = user.secondary_groups_name
+        return names unless names.empty?
+
+        defaults.secondary_groups
+      end
+
+      # Command to modify the user
+      #
+      # @param new_user [User] User containing the updated information
+      # @param old_user [User] Original user
+      # @return [Array<String>] usermod options
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/PerceivedComplexity
+      def usermod_options(new_user, old_user)
+        args = []
+        args << "--gid" << new_user.gid if new_user.gid != old_user.gid && new_user.gid
+        args << "--comment" << new_user.gecos.join(",") if new_user.gecos != old_user.gecos
+        if new_user.home != old_user.home && new_user.home
+          args << "--home" << new_user.home << "--move-home"
+        end
+        args << "--shell" << new_user.shell if new_user.shell != old_user.shell && new_user.shell
+        if different_groups?(new_user, old_user)
+          args << "--groups" << new_user.secondary_groups_name.join(",")
+        end
+        args << new_user.name
+        args
+      end
+      # rubocop:enable Metrics/CyclomaticComplexity
+      # rubocop:enable Metrics/AbcSize
+      # rubocop:enable Metrics/PerceivedComplexity
+
+      # Whether the users have different groups
+      #
+      # @param user1 [User]
+      # @param user2 [User]
+      #
+      # @return [Boolean]
+      def different_groups?(user1, user2)
+        sorted_groups(user1) != sorted_groups(user2)
+      end
+
+      # Groups of a user, sorted by id
+      #
+      # @param user [User]
+      # @return [Array<Group>]
+      def sorted_groups(user)
+        user.groups(with_primary: false).sort_by(&:id)
+      end
+
+      # Options for `useradd` to create the home directory
+      #
+      # @param _user [User]
+      # @return [Array<String>]
+      def create_home_options(_user)
+        # TODO: "--btrfs-subvolume-home" if needed
+        opts = ["--create-home"]
+        opts.concat(["--skel", defaults.skel]) if defaults.forced_skel?
```

Comment:
just style matter but it is inconsistent with rest of file that use `<<` approach for adding options.

Reply:
I will change it